source/distributed/shared_tria.cc

```diff
@@ -76,14 +103,85 @@ namespace parallel
                   active_halo_layer.find(cell) == active_halo_layer.end())
                 cell->set_subdomain_id(numbers::artificial_subdomain_id);
             }
+
+          // loop over all cells in multigrid hierarchy and mark artificial:
+          if (settings & construct_multigrid_hierarchy)
```

Comment:
It should be possible to make it work without that setting, but I don't see a good reason for allowing that. ``allow_artificial`` is more like a backwards-compatibility thing, right?